pwndbg/emu/emulator.py

```diff
@@ -171,6 +171,13 @@ def debug(debug_type, fmt, args=()) -> None:
     U.UC_ARCH_RISCV: [C.riscv_const.RISCV_INS_ECALL],
 }
 
+# We stop emulation when hitting these instructions, since they depend on co-processors or other information
+# unavailable to the emulator
+BANNED_INSTRUCTIONS = {
+    "mips": {C.mips.MIPS_INS_RDHWR},
+    "arm": {C.arm.ARM_INS_MRC, C.arm.ARM_INS_MRRC, C.arm.ARM_INS_MRC2, C.arm.ARM_INS_MRRC2},
+}
```

Comment:
Unicorn will emulate through these instructions, but get the incorrect results. They relate to receiving some value from a "coprocessor" and placing it into a register.

Fun fact: In MIPS, the `rdhwr` is seen often querying register number 29 which returns the thread local storage base: https://s3-eu-west-1.amazonaws.com/downloads-mips/documents/MIPS_Architecture_MIPS64_InstructionSet_%20AFP_P_MD00087_06.05.pdf#page=390

Likewise, MRC on Arm is used to read the TLS base.

The emulator just reads zeros into the registers, causing the subsequent instruction to be inaccurate.